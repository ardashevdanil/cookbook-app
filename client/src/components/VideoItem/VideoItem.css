.VideoItem {
	width: 312px;
	padding: 20px calc( (100% - 312px * 3)/6 );
}

.VideoItem__video {
  height: 190px;
  overflow: hidden;
}

.VideoItem img {
	width: 100%;
}

@media only screen and (max-width: 1120px) {
  .VideoItem {
    width: 100%;
    padding-top: 0;
  }

  .VideoItem>img {
    height: 64.7vw;
  }
}