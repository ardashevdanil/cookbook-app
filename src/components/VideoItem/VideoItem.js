import PropTypes from 'prop-types';
import React from 'react';
import YouTube from 'react-youtube';
import './VideoItem.css';

function VideoItem(props) {

	return (
		<div className='VideoItem'>
			<div className ='VideoItem__video'>
				<YouTube
					videoId={ props.content.videoId }
					opts={{
						height: '100%',
						width: '100%'
					}}
				/>
			</div>
		</div>
	);
}

export default VideoItem;

VideoItem.propTypes = {
	content: PropTypes.shape({		
		name: PropTypes.string.isRequired,
		videoId: PropTypes.string.isRequired,
	})
}