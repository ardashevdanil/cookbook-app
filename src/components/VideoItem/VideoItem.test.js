import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import VideoItem from './VideoItem';

Enzyme.configure({ adapter: new Adapter() });

describe('test VideoItem component', () => {
  const pushCart = jest.fn();
	const component = Enzyme.shallow(
		<VideoItem
      content={{
        name: 'test',
        videoId: '',
      }}
    />
	);

	test('renders correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});

  /*
  test('calls pushCart after click on the cart button', () => {
    component.find('div').at(6).simulate('click');

    expect(pushCart.mock.calls.length).toBe(1);
  })
  */
})