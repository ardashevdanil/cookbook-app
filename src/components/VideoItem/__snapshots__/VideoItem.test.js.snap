// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test VideoItem component renders correctly 1`] = `
"<div className=\\"VideoItem\\">
  <div className=\\"VideoItem__video\\">
    <YouTube videoId=\\"\\" opts={{...}} id={{...}} className={{...}} containerClassName=\\"\\" onReady={[Function: onReady]} onError={[Function: onError]} onPlay={[Function: onPlay]} onPause={[Function: onPause]} onEnd={[Function: onEnd]} onStateChange={[Function: onStateChange]} onPlaybackRateChange={[Function: onPlaybackRateChange]} onPlaybackQualityChange={[Function: onPlaybackQualityChange]} />
  </div>
</div>"
`;
